Decouple k8s client heartbeat from connection timeout and keep alive

dlapcevic opened the pull request:
The main reason to introduce this is to be able to increase the heartbeat interval without affecting the k8s client connection settings.

It also allows the possibility to disable heartbeat, by setting `k8s-heartbeat-timeout` to 0, without disable the k8s client itself.

```release-note
When upgrading, users can experience a change to their configuration if they were overriding the `k8s-heartbeat-timeout` flag.
K8s client timeout and keep alive are no longer getting values from the `k8s-heartbeat-timeout` flag, but have default values (30 seconds).
```

Signed-off-by: Dorde Lapcevic <[dordel@google.com](mailto:dordel@google.com)>

tommyp1ckles commented on pkg/k8s/client/config.go:
```diff
@@ -55,6 +63,8 @@ func (def Config) Flags(flags *pflag.FlagSet) {
 	flags.Float32(option.K8sClientQPSLimit, def.K8sClientQPS, "Queries per second limit for the K8s client")
 	flags.Int(option.K8sClientBurst, def.K8sClientBurst, "Burst value allowed for the K8s client")
+	flags.Duration(option.K8sClientConnTimeout, def.K8sClientConnTimeout, "Configures the timeout of K8s client connections")
+	flags.Duration(option.K8sClientConnKeepAlive, def.K8sClientConnKeepAlive, "Configures the keep alive duration of K8s client connections")
```
Can we mention that setting zero disables both of these?

dlapcevic replied:
Done. Also updated the docs.

dlapcevic committed:
Decouple k8s client heartbeat from connection timeout and keep alive

The main reason to introduce this is to be able to increase the heartbeat interval without affecting the k8s client connection settings.

It also allows the possibility to disable heartbeat, by setting `k8s-heartbeat-timeout` to 0, without disable the k8s client itself.

```release-note
When upgrading, users can experience a change to their configuration if they were overriding the `k8s-heartbeat-timeout` flag. k8s client timeout and keep alive are no longer getting values from the `k8s-heartbeat-timeout` flag, but rather would have default values (30 seconds).
```

Signed-off-by: Dorde Lapcevic <dordel@google.com>

dlapcevic committed:
Documentation for decoupling k8s-heartbeat-timeout

Signed-off-by: Dorde Lapcevic <dordel@google.com>
